client/web/src/user/settings/auth/ExternalAccountsSignIn.tsx

```diff
@@ -65,35 +101,46 @@ const getNormalizedAccount = (
     if (account && accountExternalData) {
         switch (kind) {
-            case 'GITHUB':
-                {
-                    const githubExternalData = accountExternalData as GitHubExternalData
-                    normalizedAccount = {
-                        ...normalizedAccount,
-                        external: {
-                            id: account.id,
-                            // map github fields
-                            userName: githubExternalData.name,
-                            userLogin: githubExternalData.login,
-                            userUrl: githubExternalData.html_url,
-                        },
-                    }
+            case 'github': {
+                const githubExternalData = accountExternalData as GitHubExternalData
+                normalizedAccount = {
+                    ...normalizedAccount,
+                    external: {
+                        id: account.id,
+                        // map GitHub fields
+                        userName: githubExternalData.name,
+                        userLogin: githubExternalData.login,
+                        userUrl: githubExternalData.html_url,
+                    },
                 }
+            }
                 break
-            case 'GITLAB':
-                {
-                    const gitlabExternalData = accountExternalData as GitLabExternalData
-                    normalizedAccount = {
-                        ...normalizedAccount,
-                        external: {
-                            id: account.id,
-                            // map gitlab fields
-                            userName: gitlabExternalData.name,
-                            userLogin: gitlabExternalData.username,
-                            userUrl: gitlabExternalData.web_url,
-                        },
-                    }
+            case 'gitlab': {
+                const gitlabExternalData = accountExternalData as GitLabExternalData
+                normalizedAccount = {
+                    ...normalizedAccount,
+                    external: {
+                        id: account.id,
+                        // map gitlab fields
+                        userName: gitlabExternalData.name,
+                        userLogin: gitlabExternalData.username,
+                        userUrl: gitlabExternalData.web_url,
+                    },
+                }
+            }
+                break
+            case 'saml': {
+                const samlExternalData = accountExternalData as SamlExternalData
+                // In case the SAML values don't have a username, we get the user email.
+                const userNameOrEmail = getUsernameOrEmail(samlExternalData)
+                normalizedAccount = {
+                    ...normalizedAccount,
+                    external: {
+                        id: account.id,
+                        userName: userNameOrEmail
+                    },
```

Comment:
Nit - we can inline the `userNameOrEmail` variable, since it's only used once.
```suggestion
                normalizedAccount = {
                    ...normalizedAccount,
                    external: {
                        id: account.id,
                        userName: getUsernameOrEmail(samlExternalData)
                    },
```